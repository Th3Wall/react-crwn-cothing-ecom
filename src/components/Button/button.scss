.button {
    min-width: 165px;
    width: auto;
    height: 50px;
    letter-spacing: 0.5px;
    line-height: 50px;
    padding: 0 35px 0 35px;
    font-size: 15px;
    background-color: black;
    color: white;
    text-transform: uppercase;
    font-family: 'Open Sans Condensed';
    font-weight: bolder;
    border: 1px solid transparent;
    cursor: pointer;
    transition: all 300ms cubic-bezier(0.25, 0.45, 0.45, 0.95);

    &:hover {
        background-color: white;
        color: black;
        border: 1px solid black;
    }

    &.google-sign-in {
        background-color: #4285f4;
        color: white;
        margin: 0 10px;

        &:hover {
            background-color: #2876f9;
            border: 1px solid transparent;
        }
    }
}