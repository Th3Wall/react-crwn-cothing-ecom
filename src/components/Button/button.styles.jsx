import styled, { css } from 'styled-components';

const baseButtonStyles = css`
    background-color: black;
    color: white;

    &:hover {
        background-color: white;
        color: black;
        border: 1px solid black;
    }
`

export const googleButtonStyles = css`
    background-color: #4285f4;
    color: white;
    margin: 0 10px;

    &:hover {
        background-color: #2876f9;
        border: 1px solid transparent;
    }
`

export const invertedButtonStyles = css`
    background-color: white;
    color: black;
    border: 1px solid black;

    &:hover {
        background-color: black;
        color: white;
        border: 1px solid transparent;
    }
`

const getButtonStyles = props => {
    if (props.isGoogleSignIn) return googleButtonStyles;

    return props.inverted ? invertedButtonStyles : baseButtonStyles;
}

export const BaseButton = styled.button`
    min-width: 165px;
    width: auto;
    height: 50px;
    letter-spacing: 0.5px;
    line-height: 50px;
    padding: 0 35px 0 35px;
    font-size: 15px;
    border: 1px solid transparent;
    text-transform: uppercase;
    font-family: 'Open Sans Condensed';
    font-weight: bolder;
    cursor: pointer;
    display: flex;
    justify-content: center;
    transition: all 300ms cubic-bezier(0.25, 0.45, 0.45, 0.95);
    ${getButtonStyles}
`