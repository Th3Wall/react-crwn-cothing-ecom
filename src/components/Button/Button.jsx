import './button.scss';

const Button = ({ children, isGoogleSignIn, ...otherProps }) => {
    return (
        <button
            className={`${isGoogleSignIn ? 'google-sign-in' : ''} button`}
            {...otherProps}
        >
            {children}
        </button>
    )
}

export default Button
