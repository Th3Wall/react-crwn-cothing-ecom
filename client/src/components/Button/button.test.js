import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
    it('should display Button component', () => {
        const wrapper = shallow(<Button/>);
        expect(wrapper).toMatchSnapshot();
    })
})