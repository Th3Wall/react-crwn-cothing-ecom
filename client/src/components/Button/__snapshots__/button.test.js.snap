// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should display Button component 1`] = `<styled.button />`;
